### PR TITLE
MES 6419 Cat B State

### PR DESCRIPTION
**Pull request by Cullima**
## Description
- Migrates  and updates NGRX state for Category B

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+);
+
+export const ShowMeQuestionSeriousFault = createAction(
+  '[Vehicle Checks] Show me question serious fault',
+);
+
+export const ShowMeQuestionDangerousFault = createAction(
+  '[Vehicle Checks] Show me question dangerous fault',
+);
+// export const SHOW_ME_QUESTION_REMOVE_FAULT = '[Vehicle Checks] Show me question remove fault';
```
Comment (tomcrawleyy) [inline src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts]: this looks leftover

File: src/store/tests/test-data/cat-b/test-data.cat-b.selector.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+};
+
+export const getShowMeQuestion = (state: CatBUniqueTypes.VehicleChecks) => {
+  const questionProvider: QuestionProvider = new QuestionProvider();
+  return questionProvider
+    .getShowMeQuestions(TestCategory.B)
+    .find((question) => question.code === get(state, 'showMeQuestion.code'));
+};
+
+// TODO - We should really pass a Vehicle Checks object here and not Test Data
```
Comment (tomcrawleyy) [inline src/store/tests/test-data/cat-b/test-data.cat-b.selector.ts:75]: can see that this comment was in mes app so has been brought over, however, would there be a case for addressing it here?

File: src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { createAction } from '@ngrx/store';
+import { VehicleChecksQuestion } from '../../../../../app/providers/question/vehicle-checks-question.model';
```
Comment (matthew2564) [inline src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts]: can be changed to '@providers...' with new path aliases

File: src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
+import { CompetencyOutcome } from 'src/app/shared/models/competency-outcome';
```
Comment (matthew2564) [inline src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.reducer.ts]: can be changed to '@shared...' with new path alias you have added

File: src/store/tests/test-data/cat-b/test-data.cat-b.selector.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import { VehicleChecksQuestion } from 'src/app/providers/question/vehicle-checks-question.model';
+import { Competencies, LegalRequirements } from '../test-data.constants';
+
+export const getDrivingFaultCount = (
+  data: CatBUniqueTypes.TestData, competency: Competencies,
+) => data.drivingFaults[competency];
+
+export const getManoeuvres = (data: CatBUniqueTypes.TestData): CatBUniqueTypes.Manoeuvres => data.manoeuvres;
+
+// TODO - We should pass a Manoeuvre object here instead of TestData
```
Comment (tomcrawleyy) [inline src/store/tests/test-data/cat-b/test-data.cat-b.selector.ts]: is there a case for getting this comment addressed here?

File: src/store/tests/test-data/cat-b/vehicle-checks/__tests__/vehicle-checks.reducer.spec.ts [hunk 6]
```diff
@@ -0,0 +1,157 @@
+import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
+import { CompetencyOutcome } from 'src/app/shared/models/competency-outcome';
+import { VehicleChecksQuestion } from 'src/app/providers/question/vehicle-checks-question.model';
```
Comment (matthew2564) [inline src/store/tests/test-data/cat-b/vehicle-checks/__tests__/vehicle-checks.reducer.spec.ts]: Path aliases can be used on above two imports

File: src/store/tests/test-data/cat-b/__tests__/test-data.cat-b.selector.spec.ts [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+  hasManoeuvreBeenCompletedCatB,
+  isTellMeQuestionSelected,
+  isTellMeQuestionCorrect,
+  isTellMeQuestionDrivingFault,
+  hasVehicleChecksBeenCompletedCatB,
+  hasEyesightTestGotSeriousFault,
+  hasEyesightTestBeenCompleted,
+} from '../test-data.cat-b.selector';
+import { Competencies } from '../../test-data.constants';
+import { cloneDeep } from 'lodash';
```
Comment (matthew2564) [inline src/store/tests/test-data/cat-b/__tests__/test-data.cat-b.selector.spec.ts]: think this should error with eslint due to import order, plus paths up top of file not using new aliasing

File: src/store/tests/tests-reducer-factory.ts [hunk 8]
```diff
@@ -22,6 +24,8 @@ export function testsReducerFactory(
   switch (category) {
     case TestCategory.ADI2:
       return testsCatADIPart2Reducer(action, state as Required<CatADI2UniqueTypes.TestResult>);
+    case TestCategory.B:
+      return testsCatBReducer(action, state as Required<CatBUniqueTypes.TestResult>)
```
Comment (matthew2564) [inline src/store/tests/tests-reducer-factory.ts]: missing semi-colon, think this is an eslint error

File: src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts (hunk 1 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts]: yes thanks removed

File: src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts (hunk 3 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions.ts]: changed

File: src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.reducer.ts (hunk 4 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-b/vehicle-checks/vehicle-checks.reducer.ts]: changed

File: src/store/tests/test-data/cat-b/vehicle-checks/__tests__/vehicle-checks.reducer.spec.ts (hunk 6 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-b/vehicle-checks/__tests__/vehicle-checks.reducer.spec.ts]: changed

File: src/store/tests/test-data/cat-b/__tests__/test-data.cat-b.selector.spec.ts (hunk 7 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-b/__tests__/test-data.cat-b.selector.spec.ts]: linting should be fixed now

File: src/store/tests/tests-reducer-factory.ts (hunk 8 above)
Comment (Cullima) [reply src/store/tests/tests-reducer-factory.ts]: resolved

File: src/store/tests/test-data/cat-b/test-data.cat-b.selector.ts (hunk 5 above)
Comment (Cullima) [reply src/store/tests/test-data/cat-b/test-data.cat-b.selector.ts]: i don't think this is an issue so have removed the comment. changing the selector would mean updating all usages and I don't see the benefit / problem

**Review by matthew2564 — APPROVED** [review]

**Review by tomcrawleyy — APPROVED** [review]

**Pull request merged**